test/topology/test_topology.py

```diff
@@ -154,3 +155,14 @@ async def do_remove_node():
         stopped = True
         await ddl_task
         logger.debug("ddl fiber done, finished")
+
+
+@pytest.mark.asyncio
+async def test_remove_node_group0_failure(manager):
+    """Remove a node injecting an error to fail when removing from group0"""
+    servers = await manager.running_servers()
+    await manager.server_stop_gracefully(servers[1].server_id)              # stop   1
+    async with inject_error(manager.api, servers[0].ip_addr, 'removenode_group0_after_finish',
+                            one_shot=True):
+        with pytest.raises(HTTPError, match = "node_ops_cmd_check: Node \S+ rejected"):
```

Comment:
So what's the purpose of this test? To ensure that we preserve the broken behavior? :D

Reply:
Good point. Modified to `xfail` as requested in chat.